src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -0,0 +1,177 @@
+                {
+                    rtxPairs.put(
+                        groupSources.get(0).getSSRC(),
+                        groupSources.get(1).getSSRC());
+                }
+            }
+
+            if (!simGroups.isEmpty())
+            {
+                for (SourceGroupPacketExtension simGroup : simGroups)
+                {
+                    List<SourcePacketExtension> simSources
+                        = simGroup.getSources();
+
+                    RTPEncodingImpl[] rtpEncodings
+                        = new RTPEncodingImpl[simSources.size() * 3];
+                    MediaStreamTrackImpl track = new MediaStreamTrackImpl(
+                        mediaStreamTrackReceiver, rtpEncodings);
+
+                    for (int i = 0; i < simSources.size(); i++)
+                    {
+                        SourcePacketExtension spe = simSources.get(i);
+                        long primarySSRC = spe.getSSRC();
+                        grouped.add(primarySSRC);
+
+                        long rtxSSRC = -1;
+                        if (rtxPairs.containsKey(primarySSRC))
+                        {
+                            rtxSSRC = rtxPairs.remove(primarySSRC);
+                            grouped.add(rtxSSRC);
+                        }
+
+                        int subjectiveQualityIdx = i * 3;
+                        RTPEncodingImpl encoding = new RTPEncodingImpl(
+                            track, subjectiveQualityIdx, primarySSRC, rtxSSRC,
+                            0, null);
+
+                        rtpEncodings[subjectiveQualityIdx] = encoding;
+
+                        subjectiveQualityIdx = i * 3 + 1;
+                        encoding = new RTPEncodingImpl(
+                            track, subjectiveQualityIdx, primarySSRC, rtxSSRC,
+                            1, new RTPEncodingImpl[] { encoding });
+                        rtpEncodings[subjectiveQualityIdx] = encoding;
+
+                        subjectiveQualityIdx = i * 3 + 2;
+                        encoding = new RTPEncodingImpl(
+                            track, subjectiveQualityIdx, primarySSRC, rtxSSRC,
+                            2, new RTPEncodingImpl[] { encoding });
+                        rtpEncodings[subjectiveQualityIdx] = encoding;
+                    }
+
+                    tracks.add(track);
+                }
+            }
+
+            if (!rtxPairs.isEmpty())
+            {
+                for (Map.Entry<Long, Long> fidEntry : rtxPairs.entrySet())
+                {
+                    Long primarySSRC = fidEntry.getKey();
+                    Long rtxSSRC = fidEntry.getValue();
+
+                    RTPEncodingImpl[] encodings = new RTPEncodingImpl[1];
+                    MediaStreamTrackImpl track = new MediaStreamTrackImpl(
+                        mediaStreamTrackReceiver, encodings);
+                    encodings[0]
+                        = new RTPEncodingImpl(track, primarySSRC, rtxSSRC);
```

Comment:
I'm still trying to wrap my head around this. Here we have a (SSRC, rtxSSRC) pair outside any simulcast groups. So this corresponds to a new track. We choose to not handle temporal layers, so the track has a single RTPEncoding.
In this case, this RTPEncoding must match all packets for its SSRC (and RTX SSRC), regardless of their temporal layer. However, the way RTPEncodingImpl is initialized, it ends up having temporalId=0, which means that it only accepts packets with tid=0, or no tid. If for some reason the remote side decides to encode its stream with temporal layers, the RTPEncoding will miss some of the packets. Now, that's a bit of a corner case, but I think there's an easy and intuitive way of handling it by using an explicit "wildcard" value for the RTPEncoding's temporalId value (i.e. if the RTPEncoding has temporalId = -1, then it skips the tid check for packets altogether).